[GTK] Add a DisplayLink implementation

carlosgcampos opened the pull request:
#### 4aad349224a3cf6246969fef86972f191b280fc1
<pre>
[GTK] Add a DisplayLink implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=261673">https://bugs.webkit.org/show_bug.cgi?id=261673</a>

Reviewed by Simon Fraser.

Add DisplayLink implementation using DRM API when available to get the
vblank signal from monitor. When DRM is not available we fallback to use
a timer. The patch includes some refactorings needed to share the
implementation with other ports for non-platform specific parts. The
platform display ID is no longer generated based on the drawing area
identifier, but assigned by the screen manager to every monitor.
ThreadedDisplayRefreshMonitor and DisplayRefreshMonitorGtk are no longer
needed now.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/graphics/DisplayRefreshMonitor.cpp:
(WebCore::DisplayRefreshMonitor::createDefaultDisplayRefreshMonitor):
(WebCore::DisplayRefreshMonitor::requestRefreshCallback):
* Source/WebCore/platform/graphics/gtk/DisplayRefreshMonitorGtk.cpp: Removed.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::frameComplete):
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseUpdateDisplayID):
(toplevelWindowConfigureEvent):
(webkitWebViewBaseSetToplevelOnScreenWindow):
(webkitWebViewBaseRealize):
(toplevelWindowMonitorsChanged):
(toplevelWindowRealized):
(toplevelWindowUnrealized):
(webkitWebViewBaseRoot):
(webkitWebViewBaseUnroot):
(webkitWebViewBaseCreateWebPage):
(webkitWebViewBaseDidRelaunchWebProcess):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::displayNominalFramesPerSecond):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/DisplayLink.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLink.cpp.
(WebKit::DisplayLink::DisplayLink):
(WebKit::DisplayLink::~DisplayLink):
(WebKit::DisplayLink::addObserver):
(WebKit::DisplayLink::notifyObserversDisplayDidRefresh):
(WebKit::DisplayLinkCollection::stopDisplayLinks):
* Source/WebKit/UIProcess/DisplayLink.h: Renamed from Source/WebKit/UIProcess/mac/DisplayLink.h.
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.cpp.
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.h: Copied from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.h.
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.h:
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.cpp:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::Internals::Internals):
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::updateWheelEventActivityAfterProcessSwap):
(WebKit::WebPageProxy::windowScreenDidChange):
(WebKit::WebPageProxy::setHasActiveAnimatedScrolls):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::connectionWillOpen):
(WebKit::WebProcessProxy::processWillShutDown):
(WebKit::WebProcessProxy::nominalFramesPerSecondForDisplay):
(WebKit::WebProcessProxy::startDisplayLink):
(WebKit::WebProcessProxy::stopDisplayLink):
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond):
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/UIProcess/glib/DisplayLinkGLib.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.h.
(WebKit::DisplayLink::platformInitialize):
(WebKit::DisplayLink::platformFinalize):
(WebKit::DisplayLink::platformIsRunning const):
(WebKit::DisplayLink::platformStart):
(WebKit::DisplayLink::platformStop):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitor.cpp: Added.
(WebKit::DisplayVBlankMonitor::create):
(WebKit::DisplayVBlankMonitor::DisplayVBlankMonitor):
(WebKit::DisplayVBlankMonitor::~DisplayVBlankMonitor):
(WebKit::DisplayVBlankMonitor::startThreadIfNeeded):
(WebKit::DisplayVBlankMonitor::start):
(WebKit::DisplayVBlankMonitor::stop):
(WebKit::DisplayVBlankMonitor::invalidate):
(WebKit::DisplayVBlankMonitor::isActive):
(WebKit::DisplayVBlankMonitor::setHandler):
(WebKit::DisplayVBlankMonitor::destroyThreadTimerFired):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitor.h: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::DisplayVBlankMonitor::refreshRate const):
(WebKit::DisplayVBlankMonitor::WTF_GUARDED_BY_LOCK):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp: Added.
(WebKit::findCrtc):
(WebKit::DisplayVBlankMonitorDRM::create):
(WebKit::DisplayVBlankMonitorDRM::DisplayVBlankMonitorDRM):
(WebKit::DisplayVBlankMonitorDRM::waitForVBlank const):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.h: Renamed from Source/WebCore/platform/graphics/gtk/DisplayRefreshMonitorGtk.h.
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.cpp: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::DisplayVBlankMonitorTimer::create):
(WebKit::DisplayVBlankMonitorTimer::DisplayVBlankMonitorTimer):
(WebKit::DisplayVBlankMonitorTimer::waitForVBlank const):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.h: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
* Source/WebKit/UIProcess/gtk/ScreenManager.cpp:
(WebKit::ScreenManager::displayID const):
(WebKit::ScreenManager::monitor const):
* Source/WebKit/UIProcess/gtk/ScreenManager.h:
* Source/WebKit/UIProcess/mac/DisplayLinkMac.cpp: Added.
(WebKit::DisplayLink::platformInitialize):
(WebKit::DisplayLink::platformFinalize):
(WebKit::DisplayLink::nominalFramesPerSecondFromDisplayLink):
(WebKit::DisplayLink::platformIsRunning const):
(WebKit::DisplayLink::platformStart):
(WebKit::DisplayLink::platformStop):
(WebKit::DisplayLink::displayLinkCallback):
* Source/WebKit/UIProcess/mac/WebProcessProxyMac.mm:
(WebKit::WebProcessProxy::nominalFramesPerSecondForDisplay): Deleted.
(WebKit::WebProcessProxy::startDisplayLink): Deleted.
(WebKit::WebProcessProxy::stopDisplayLink): Deleted.
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond): Deleted.
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::createDisplayRefreshMonitor):
(WebKit::DrawingAreaCoordinatedGraphics::enterAcceleratedCompositingMode):
(WebKit::DrawingAreaCoordinatedGraphics::exitAcceleratedCompositingMode):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::didRenderFrame):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor):
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.h:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.messages.in:
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.cpp: Renamed from Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.cpp.
(WebKit::WebDisplayRefreshMonitor::WebDisplayRefreshMonitor):
(WebKit::WebDisplayRefreshMonitor::~WebDisplayRefreshMonitor):
(WebKit::WebDisplayRefreshMonitor::dispatchDisplayDidRefresh):
(WebKit::WebDisplayRefreshMonitor::startNotificationMechanism):
(WebKit::WebDisplayRefreshMonitor::stopNotificationMechanism):
(WebKit::WebDisplayRefreshMonitor::adjustPreferredFramesPerSecond):
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.h: Renamed from Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.h.
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::createDisplayRefreshMonitor):
* Source/WebKit/WebProcess/WebProcess.cpp:
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:

Co-authored-by: Carlos Garcia Campos &lt;cgarcia@igalia.com&gt;
Canonical link: <a href="https://commits.webkit.org/268646@main">https://commits.webkit.org/268646@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0aebc530989654653ac9d99fa00258993edd5330

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20231 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20663 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21299 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22131 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18880 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20462 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23916 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20830 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20325 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20450 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20357 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17590 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22981 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17525 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18393 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24650 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/17600 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18602 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18569 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22622 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/19602 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19148 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16259 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/23637 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18353 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5676 "Found 2 jsc stress test failures: stress/sampling-profiler-microtasks.js.default, wasm.yaml/wasm/stress/simple-inline-stacktrace-with-catch.js.wasm-eager") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4871 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22695 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/24894 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18973 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5500 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -569,7 +569,7 @@ WebPageProxy::Internals::Internals(WebPageProxy& page)
     , updateReportedMediaCaptureStateTimer(RunLoop::main(), &page, &WebPageProxy::updateReportedMediaCaptureState)
     , webPageID(PageIdentifier::generate())
-#if HAVE(CVDISPLAYLINK)
+#if HAVE(CVDISPLAYLINK) || PLATFORM(GTK)
```
There are so many of these that I think it would be useful to define `HAVE_DISPLAY_LINK` for Mac and GTK and use it in all these locations.

carlosgcampos replied:
Thanks! This is a great idea, I'll also use it instead of `PLATFORM(GTK)` for other places where we really mean `HAVE(DISPLAY_LINK)` and will make the patch to add wpe display link support simpler too.

alexgcastro committed:
[GTK] Add a DisplayLink implementation
https://bugs.webkit.org/show_bug.cgi?id=261673

Reviewed by Simon Fraser.

Add DisplayLink implementation using DRM API when available to get the
vblank signal from monitor. When DRM is not available we fallback to use
a timer. The patch includes some refactorings needed to share the
implementation with other ports for non-platform specific parts. The
platform display ID is no longer generated based on the drawing area
identifier, but assigned by the screen manager to every monitor.
ThreadedDisplayRefreshMonitor and DisplayRefreshMonitorGtk are no longer
needed now.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/graphics/DisplayRefreshMonitor.cpp:
(WebCore::DisplayRefreshMonitor::createDefaultDisplayRefreshMonitor):
(WebCore::DisplayRefreshMonitor::requestRefreshCallback):
* Source/WebCore/platform/graphics/gtk/DisplayRefreshMonitorGtk.cpp: Removed.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::frameComplete):
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseUpdateDisplayID):
(toplevelWindowConfigureEvent):
(webkitWebViewBaseSetToplevelOnScreenWindow):
(webkitWebViewBaseRealize):
(toplevelWindowMonitorsChanged):
(toplevelWindowRealized):
(toplevelWindowUnrealized):
(webkitWebViewBaseRoot):
(webkitWebViewBaseUnroot):
(webkitWebViewBaseCreateWebPage):
(webkitWebViewBaseDidRelaunchWebProcess):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::displayNominalFramesPerSecond):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/DisplayLink.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLink.cpp.
(WebKit::DisplayLink::DisplayLink):
(WebKit::DisplayLink::~DisplayLink):
(WebKit::DisplayLink::addObserver):
(WebKit::DisplayLink::notifyObserversDisplayDidRefresh):
(WebKit::DisplayLinkCollection::stopDisplayLinks):
* Source/WebKit/UIProcess/DisplayLink.h: Renamed from Source/WebKit/UIProcess/mac/DisplayLink.h.
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.cpp.
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.h: Copied from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.h.
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.h:
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.cpp:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::Internals::Internals):
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::updateWheelEventActivityAfterProcessSwap):
(WebKit::WebPageProxy::windowScreenDidChange):
(WebKit::WebPageProxy::setHasActiveAnimatedScrolls):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::connectionWillOpen):
(WebKit::WebProcessProxy::processWillShutDown):
(WebKit::WebProcessProxy::nominalFramesPerSecondForDisplay):
(WebKit::WebProcessProxy::startDisplayLink):
(WebKit::WebProcessProxy::stopDisplayLink):
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond):
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/UIProcess/glib/DisplayLinkGLib.cpp: Renamed from Source/WebKit/UIProcess/mac/DisplayLinkProcessProxyClient.h.
(WebKit::DisplayLink::platformInitialize):
(WebKit::DisplayLink::platformFinalize):
(WebKit::DisplayLink::platformIsRunning const):
(WebKit::DisplayLink::platformStart):
(WebKit::DisplayLink::platformStop):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitor.cpp: Added.
(WebKit::DisplayVBlankMonitor::create):
(WebKit::DisplayVBlankMonitor::DisplayVBlankMonitor):
(WebKit::DisplayVBlankMonitor::~DisplayVBlankMonitor):
(WebKit::DisplayVBlankMonitor::startThreadIfNeeded):
(WebKit::DisplayVBlankMonitor::start):
(WebKit::DisplayVBlankMonitor::stop):
(WebKit::DisplayVBlankMonitor::invalidate):
(WebKit::DisplayVBlankMonitor::isActive):
(WebKit::DisplayVBlankMonitor::setHandler):
(WebKit::DisplayVBlankMonitor::destroyThreadTimerFired):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitor.h: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::DisplayVBlankMonitor::refreshRate const):
(WebKit::DisplayVBlankMonitor::WTF_GUARDED_BY_LOCK):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp: Added.
(WebKit::findCrtc):
(WebKit::DisplayVBlankMonitorDRM::create):
(WebKit::DisplayVBlankMonitorDRM::DisplayVBlankMonitorDRM):
(WebKit::DisplayVBlankMonitorDRM::waitForVBlank const):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.h: Renamed from Source/WebCore/platform/graphics/gtk/DisplayRefreshMonitorGtk.h.
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.cpp: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::DisplayVBlankMonitorTimer::create):
(WebKit::DisplayVBlankMonitorTimer::DisplayVBlankMonitorTimer):
(WebKit::DisplayVBlankMonitorTimer::waitForVBlank const):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.h: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
* Source/WebKit/UIProcess/gtk/ScreenManager.cpp:
(WebKit::ScreenManager::displayID const):
(WebKit::ScreenManager::monitor const):
* Source/WebKit/UIProcess/gtk/ScreenManager.h:
* Source/WebKit/UIProcess/mac/DisplayLinkMac.cpp: Added.
(WebKit::DisplayLink::platformInitialize):
(WebKit::DisplayLink::platformFinalize):
(WebKit::DisplayLink::nominalFramesPerSecondFromDisplayLink):
(WebKit::DisplayLink::platformIsRunning const):
(WebKit::DisplayLink::platformStart):
(WebKit::DisplayLink::platformStop):
(WebKit::DisplayLink::displayLinkCallback):
* Source/WebKit/UIProcess/mac/WebProcessProxyMac.mm:
(WebKit::WebProcessProxy::nominalFramesPerSecondForDisplay): Deleted.
(WebKit::WebProcessProxy::startDisplayLink): Deleted.
(WebKit::WebProcessProxy::stopDisplayLink): Deleted.
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond): Deleted.
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::createDisplayRefreshMonitor):
(WebKit::DrawingAreaCoordinatedGraphics::enterAcceleratedCompositingMode):
(WebKit::DrawingAreaCoordinatedGraphics::exitAcceleratedCompositingMode):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::didRenderFrame):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor):
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.h:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.messages.in:
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.cpp: Renamed from Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.cpp.
(WebKit::WebDisplayRefreshMonitor::WebDisplayRefreshMonitor):
(WebKit::WebDisplayRefreshMonitor::~WebDisplayRefreshMonitor):
(WebKit::WebDisplayRefreshMonitor::dispatchDisplayDidRefresh):
(WebKit::WebDisplayRefreshMonitor::startNotificationMechanism):
(WebKit::WebDisplayRefreshMonitor::stopNotificationMechanism):
(WebKit::WebDisplayRefreshMonitor::adjustPreferredFramesPerSecond):
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.h: Renamed from Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.h.
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::createDisplayRefreshMonitor):
* Source/WebKit/WebProcess/WebProcess.cpp:
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:

Co-authored-by: Carlos Garcia Campos <cgarcia@igalia.com>
Canonical link: https://commits.webkit.org/268646@main
